script/DeployContracts.s.sol

```diff
@@ -20,60 +22,107 @@ import {BitpandaEcosystemTokenDeployer} from "./helpers/BitpandaEcosystemTokenDe
- *     <bestSupply> <nextTransferId> <otherValidators>
+ *     --sender <sender> --rpc-url <rpc alias> --slow --force --sig \
+ *     "run(address,address,address,address,address,uint256,uint256,uint256,address[])" \
+ *     <validator> <deployer> <pauser> <mediumCriticalOps> <superCriticalOps> \
+ *     <panSupply> <bestSupply> <nextTransferId> <otherValidators>
  */
 contract DeployContracts is
     PantosHubDeployer,
     PantosForwarderDeployer,
     PantosWrapperDeployer,
     PantosTokenDeployer,
-    BitpandaEcosystemTokenDeployer
+    BitpandaEcosystemTokenDeployer,
+    AccessControllerDeployer
 {
     IPantosHub public pantosHubProxy;
     PantosForwarder public pantosForwarder;
     PantosToken public pantosToken;
     BitpandaEcosystemToken public bitpandaEcosystemToken;
+    AccessController public accessController;
+    address public deployer;
+    address public pauser;
+    address public mediumCriticalOps;
+    address public superCriticalOps;
 
     function exportContractAddresses() public {
         string memory blockchainName = determineBlockchain().name;
         string memory addresses;
         for (uint256 i; i < pantosWrappers.length; i++) {
             vm.serializeAddress(
-                addresses,
+                "addresses",
                 pantosWrappers[i].symbol(),
                 address(pantosWrappers[i])
             );
         }
 
-        vm.serializeAddress(addresses, "hub_proxy", address(pantosHubProxy));
-        vm.serializeAddress(addresses, "forwarder", address(pantosForwarder));
-        vm.serializeAddress(addresses, "pan", address(pantosToken));
+        vm.serializeAddress("addresses", "hub_proxy", address(pantosHubProxy));
+        vm.serializeAddress(
+            "addresses",
+            "forwarder",
+            address(pantosForwarder)
+        );
+        vm.serializeAddress("addresses", "pan", address(pantosToken));
+        vm.serializeAddress(
+            "addresses",
+            "access_controller",
+            address(accessController)
+        );
         addresses = vm.serializeAddress(
-            addresses,
+            "addresses",
             "best",
             address(bitpandaEcosystemToken)
         );
         vm.writeJson(addresses, string.concat(blockchainName, ".json"));
     }
 
+    function exportPantosRolesAddresses() public {
+        string memory blockchainName = determineBlockchain().name;
+        string memory roles;
+        vm.serializeAddress("roles", "deployer", deployer);
+        vm.serializeAddress("roles", "pauser", pauser);
+        vm.serializeAddress("roles", "medium_critical_ops", mediumCriticalOps);
+        roles = vm.serializeAddress(
+            "roles",
+            "super_critical_ops",
+            superCriticalOps
+        );
+        vm.writeJson(roles, string.concat(blockchainName, "_ROLES.json"));
+    }
+
     function run(
         address primaryValidator,
+        address _deployer,
```

Comment:
We usually use trailing instead of leading underscores for parameter name conflicts.